libsolutil/hyperlink.h

```diff
@@ -0,0 +1,72 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <boost/filesystem.hpp>
+
+#include <iostream>
+#include <string>
+
+// Required for gethostname()
+#if defined(_WIN32)
+#include <Winsock2.h>
+#else
+#include <unistd.h>
+#endif
+
+namespace solidity::util
+{
+
+struct Hyperlink
+{
+	bool enabled = false;
+
+	struct Ref { bool enabled; std::string text; };
+
+	Ref operator()(std::string const& _ref) { return Ref{enabled, _ref}; }
+};
+
+inline std::ostream& operator<<(std::ostream& _os, Hyperlink::Ref const& _hyperlink)
+{
+	auto const path = boost::filesystem::path{_hyperlink.text};
+	bool const candidate = boost::filesystem::is_regular_file(path) && (&_os == &std::cout || &_os == &std::cerr);
+
+	if (_hyperlink.enabled && candidate)
+	{
+		static std::string const hostname = []() -> std::string {
+			char hostname[80] = {0};
+			if (gethostname(hostname, sizeof(hostname)) < 0)
+				return std::string{};
+			return std::string{hostname};
+		}();
+
+		auto const abspath = boost::filesystem::canonical(boost::filesystem::absolute(path));
+		auto constexpr OSC8 = "\033]8;;";
+		auto constexpr ST = "\033\\";
+
+		_os << OSC8 << "file://" << hostname << abspath.generic_string() << ST;
```

Comment:
hostname?

Reply:
this is the hostname from line 51, it is recommended to add a hostname to the hyperlink anchor as someone might want to call the compiler (/tool) via SSH from a remote host, and the local output then of course should know about what host this file is coming from.

Reply:
Please keep any networking-related things far away from the compiler. Accessing the filesystem is already bad enough. Also, I never heard that `file://`-urls work across hosts. Which protocol would a client use to access the file?

Reply:
`file://` with host name present is (I think) only helpful for detecting whether this is a resource from the host this text was generated from. If the host part doesn't match the local hostname, then it will not be hinted as a file you can click to follow.

Software that supports that (as far as I tested): any VTE powerd terminal, including GNOME terminal, mintty, kitty, iTerm2.